### PR TITLE
auditd_rules rework

### LINKED ISSUE #312: rework auditd_rules resource
A typical use-case will have multiple rules listed. If we can find a better matcher than checking for a regex and also making giving the user nice output if the rule isn't found in the list. Let's find real-life LIST_RULES examples and use them for our docs. The current format may lead to confusion.

### PR DESCRIPTION
**Pull request by srenatus**
Fixes #312 and #309, without dropping support for the old syntax. (Also documents the old `status` method.)

The new features, however, you only get with an not-outdated version of the `audit` package.

Adds a utility class for chaining filter methods that could be useful in more places, I hope.

Note that I couldn't find a way to have `MockLoader` present the outdated format, so I've just dropped that test for now.

### CODE REVIEW

File: docs/resources.rst [hunk 1]
```diff
@@ -352,12 +354,47 @@ The following examples show how to use this InSpec audit resource.
...
+
+**Query the audit daemon status.**
+
+.. code-block:: ruby
+
+   describe auditd_rules.status('backlog') do
+     it { should cmp 0 }
+   end
+
+**Query properties of rules targetting specific syscalls or files.**
```
Comment (arlimus) [inline docs/resources.rst]: `targeting` typo

Comment (srenatus) [inline docs/resources.rst]: thanks, fixed

**Comment by chef-supermarket** [comment]
Hi. I am an automated pull request bot named Curry. There are commits in this pull request whose authors are not yet authorized to contribute to Chef Software, Inc. projects or are using a non-GitHub verified email address. To become authorized to contribute, you will need to sign the Contributor License Agreement (CLA) as an individual or on behalf of your company. [You can read more on Chef's blog.](https://www.getchef.com/blog/2014/06/23/changes-to-the-contributor-license-agreement-process)

## GitHub Users Who Are Not Authorized To Contribute

The following GitHub users do not appear to have signed a CLA:

* @arlimus

[Please sign the CLA here.](https://supermarket.chef.io/become-a-contributor)

**Comment by arlimus** [comment]
this turned out to be an awesome improvement!! :)
rebased, just one more feedback from @srenatus on that last commit for semantics

**Comment by arlimus** [comment]
:+1:

**Comment by arlimus** [comment]
blocked by unit tests failing after winrm update: https://github.com/chef/inspec/pull/439

**Pull request merged**